File: tests/repo-intel/fixtures/ts-js-app/src/model/Case.ts
/** Case domain model: interface, type alias, and enum. */

export enum CaseStatus {
  Open = "open",
  Closed = "closed",
  Pending = "pending",
}

export interface Case {
  id: string;
  title: string;
  status: CaseStatus;
}

export type CaseId = string;

/** Second top-level `Case` symbol lives in helper (const) — same simple name trap. */
export const CASE_DEFAULT_TITLE = "untitled";

File: tests/repo-intel/fixtures/java-spring/src/main/java/com/acme/model/Case.java
package com.acme.model;

public class Case extends BaseEntity {
    private String title;
    private CaseStatus status;

    public Case() {
        this.status = CaseStatus.OPEN;
    }

    public Case(String title) {
        this.title = title;
        this.status = CaseStatus.OPEN;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public CaseStatus getStatus() {
        return status;
    }

    public void setStatus(CaseStatus status) {
        this.status = status;
    }
}
